statsmodels/graphics/functional.py

```diff
@@ -1,14 +1,315 @@
 """Module for functional boxplots."""
 
+from statsmodels.multivariate.pca import PCA
+from statsmodels.nonparametric.kernel_density import KDEMultivariate
 from statsmodels.compat.python import combinations, range
 import numpy as np
-from scipy import stats
 from scipy.misc import factorial
+import matplotlib.pyplot as plt
 
 from . import utils
 
 
-__all__ = ['fboxplot', 'rainbowplot', 'banddepth']
+__all__ = ['hdrboxplot', 'fboxplot', 'rainbowplot', 'banddepth']
+
+
+def kernel_smoothing_(data, optimize=False):
```

Comment:
those helper function need leading underscore in name to make them private, no trailing underscore